bin/run.sh

```diff
@@ -25,15 +27,52 @@ SLUG="$1"
 INPUT_DIR="${2%/}"
 OUTPUT_DIR="${3%/}"
 
-if [[ "${RUN_IN_DOCKER}" == "TRUE" ]]; then  
+if [[ "${RUN_IN_DOCKER}" == "TRUE" ]]; then
+    # Docker image contains a prebuilt package called TestEnvironment with a ready .build directory 
+    # to build solutions as fast as it possible. To minimize changes in the build graph,
+    # exercise resources are copied inside the TestEnvironment package as if they had always been there.
+
+    get_target_name() {
+        local section=$1
+        local path
+        path=$(jq -r --arg section "${section}" '.files.[$section][0]' "${CONFIG_FILE}")
+        path=${path%/*}
+        path=${path##*/}
+        echo "${path}"
+    }
+
     WORKING_DIR="${PWD}"
-    cp -r "${INPUT_DIR}/." "${WORKING_DIR}"
+    CONFIG_FILE="${INPUT_DIR}/.meta/config.json"
+    SOURCE_TARGET_PATH="${WORKING_DIR}/Sources/TestEnvironment"
+    TEST_TARGET_PATH="${WORKING_DIR}/Tests/TestEnvironmentTests"
+
+    # 1. Replace source files with those of an exercise, saving TestEnvironement paths
+    rm -r "${SOURCE_TARGET_PATH}" "${TEST_TARGET_PATH}"
+
+    target_name=$(get_target_name solution)
+    test_target_name=$(get_target_name test)
+
+    # Copying everything from package in case of other targets.
+    cp -r "${INPUT_DIR}/Sources/." "${WORKING_DIR}/Sources"
+    cp -r "${INPUT_DIR}/Tests/." "${WORKING_DIR}/Tests"
+
+    mv "${WORKING_DIR}/Sources/${target_name}" "${SOURCE_TARGET_PATH}"
+    mv "${WORKING_DIR}/Tests/${test_target_name}" "${TEST_TARGET_PATH}"
+
+    # 2. Replace @testable import SomeModule with @testable import TestEnvironment
+    sed -i 's/@testable import [^ ]\+/@testable import TestEnvironment/g' "${WORKING_DIR}/Tests/TestEnvironmentTests"/*.swift
+
+    # 3. Copy Package.swift and rename main & test targets.
+    cp "${INPUT_DIR}/Package.swift" "${WORKING_DIR}"
+    sed -i "s/\"${target_name}\"/\"TestEnvironment\"/g" "${WORKING_DIR}/Package.swift"
+    sed -i "s/\"${test_target_name}\"/\"TestEnvironmentTests\"/g" "${WORKING_DIR}/Package.swift"
```

Comment:
Note, you can do this with a single `sed` invocation. You only need to read and re-write the file once.